### PR TITLE
Add support for Base chain in MorphoProtocolPlugin

### PR DESCRIPTION
**Pull request by piotrwitek**
Added necessary changes to enable support for Base chain in MorphoProtocolPlugin

## Summary by CodeRabbit

- **New Features**
	- Added support for the "Base" chain in both the Morpho and Spark protocol plugins, enhancing compatibility with additional blockchain networks.
	- Introduced a new token entry for "Based ETH" (BSDETH) in the static tokens list.

- **Enhancements**
	- Updated error messages in the Morpho plugin for clearer reporting.
	- Improved type safety by incorporating new type definitions in the Morpho protocol plugin tests.
	- Adjusted test configurations to focus on the base chain instead of the mainnet.

- **Chores**
	- Modified TypeScript configuration for the build process with the introduction of a new configuration file.
	- Renamed the "Mainnet" entry to "Base" in the ChainNames enumeration for clarity.

### CODE REVIEW

File: sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts [hunk 1]
```diff
@@ -344,7 +347,7 @@ export class MorphoProtocolPlugin extends BaseProtocolPlugin {
 
     if (!debtToken) {
       throw new Error(
-        `Invalid debt token address: ${marketParameters[0]} for chain ${morphoLendingPoolId.protocol.chainInfo.name}`,
+        `Invalid debt token address: ${marketParameters[0]} for chain ${morphoLendingPoolId.protocol.chainInfo.chainId} ${morphoLendingPoolId.protocol.chainInfo.name}`,
```
Comment (robercano) [inline sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts:350]: You could also do:

```suggestion
        `Invalid debt token address: ${marketParameters[0]} for chain ${morphoLendingPoolId.protocol.chainInfo}`,
```

and `ChainInfo.toString` should take care

File: sdk/protocol-plugins/tests/unit/plugins/MorphoProtocolPlugin.spec.ts [hunk 2]
```diff
@@ -108,6 +109,8 @@ describe('Protocol Plugin | Unit | Morpho', () => {
     const positionId = MorphoLendingPositionId.createFrom({
       id: 'mockPositionId',
     })
+    const a: ILendingPositionId = positionId
```
Comment (robercano) [inline sdk/protocol-plugins/tests/unit/plugins/MorphoProtocolPlugin.spec.ts:112]: leftover?

**Review by robercano — APPROVED** [review]
Looks great! Just a suggestion and a bit of unused code, approved anyway

File: sdk/protocol-plugins/tests/unit/plugins/MorphoProtocolPlugin.spec.ts (hunk 2 above)
Comment (piotrwitek) [reply sdk/protocol-plugins/tests/unit/plugins/MorphoProtocolPlugin.spec.ts:112]: yes thanks

**Review by robercano — DISMISSED** [review]

**Pull request merged**